src-electron/generator/helper-endpointconfig.js

```diff
@@ -651,15 +651,14 @@ function determineAttributeDefaultValue(
   typeSize,
   isNullable
 ) {
-  if (specifiedDefault !== null || !isNullable) {
+  if (specifiedDefault && (specifiedDefault !== null || !isNullable)) {
```

Comment:
is `specifiedDefault !== null ` and `!isNullable` different ?

Reply:
I am confused about why you need `specifiedDefault !== null`

Reply:
not totally sure if they are always the same

Reply:
from looking at endpoint-config.c, 

`  { 0x00000000, ZAP_TYPE(TEMPERATURE), 2, ZAP_ATTRIBUTE_MASK(NULLABLE), ZAP_SIMPLE_DEFAULT(0x8000) }, /* LocalTemperature */  \`

they are different entities

Reply:
Is `specifiedDefault` not handling `specifiedDefault !== null` ?

Reply:
specifiedDefault was undefined and not null which caused the bug

Reply:
we probably don't need `specifiedDefault !== null'

Reply:
ill do a phase 2 next week

Reply:
ok the worrying part here is `specifiedDefault` can be a bit off for attributes which are boolean(false) or number(0) or String(empty) . Do not add  `specifiedDefault` to this condition unless sure what is the issue without it and how this condition solved it.

Reply:
Im confused, I didn't add anything except a check to make sure it is defined

Reply:
Example use case which can run into an issue: There is a boolean attribute and has a default value of false
in your condition if (specifiedDefault && (specifiedDefault !== null || !isNullable)) { you will not return false but return
 // Assume everything else is an unsigned integer.
  return '0x' + 'FF'.repeat(typeSize)
  
  Log a bug to revert this and thoroughly check the issue at hand